src/esp/io/json.h

```diff
@@ -39,285 +40,6 @@ std::string jsonToString(const JsonDocument& d);
- * doc for existence of passed @p tag as Magnum::Vector3. If present, populate
- * passed @p val with value. Returns whether tag is found and successfully
- * populated, or not. Logs an error if tag is found but is inappropriate type.
- *
- * @param d json document to parse
- * @param tag string tag to look for in json doc
- * @param val destination value to be populated
- * @return whether successful or not
- */
-
-template <>
-inline bool jsonIntoVal(const JsonGenericValue& d,
-                        const char* tag,
-                        Magnum::Vector3& val) {
-  if (d.HasMember(tag) && d[tag].IsArray() && d[tag].Size() == 3) {
-    for (rapidjson::SizeType i = 0; i < 3; ++i) {
-      if (d[tag][i].IsNumber()) {
-        val[i] = d[tag][i].GetDouble();
-      } else {
-        LOG(ERROR) << " Invalid numeric value specified in JSON Vec3 config at "
-                   << tag << " index :" << i;
-        return false;
-      }
-    }  // build array
-    return true;
-  }
-  return false;
-}  // jsonIntoVal<Magnum::Vector3>
-
-/**
- * @brief Specialization to handle Magnum::Quaternion values.  Check passed json
- * doc for existence of passed @p tag as Magnum::Quaternion. If present,
- * populate passed @p val with value. Returns whether tag is found and
- * successfully populated, or not. Logs an error if tag is found but is
- * inappropriate type.
- *
- * @param d json document to parse
- * @param tag string tag to look for in json doc
- * @param val destination value to be populated
- * @return whether successful or not
- */
-
-template <>
-inline bool jsonIntoVal(const JsonGenericValue& d,
-                        const char* tag,
-                        Magnum::Quaternion& val) {
-  if (d.HasMember(tag) && d[tag].IsArray() && d[tag].Size() == 4) {
-    for (rapidjson::SizeType i = 0; i < 4; ++i) {
-      if (d[tag][i].IsNumber()) {
-        if (i == 0) {
-          val.scalar() = d[tag][0].GetFloat();
-        } else {
-          val.vector()[i - 1] = d[tag][i].GetFloat();
-        }
-      } else {
-        LOG(ERROR)
-            << " Invalid numeric value specified in JSON Quaternion config at "
-            << tag << " index :" << i;
-        return false;
-      }
-    }  // build array
-    return true;
-  }
-  return false;
-}  // jsonIntoVal<Magnum::Quaternion>
-
-/**
- * @brief Specialization to handle reading a JSON object into an
- * std::map<std::string, std::string>.  Check passed json doc for existence of
- * passed @p tag and verify it is an object. If present, populate passed @p val
- * with key-value pairs in cell. Returns whether tag is found and successfully
- * populated, or not. Logs an error if tag is found but is inappropriately
- * configured
- *
- * @param d json document to parse
- * @param tag string tag to look for in json doc
- * @param val destination std::map to be populated
- * @return whether successful or not
- */
-
-template <>
-inline bool jsonIntoVal(const JsonGenericValue& d,
```

Comment:
moved to JsonStlTypes.h https://github.com/facebookresearch/habitat-sim/pull/980/files#diff-eaa098e6aba025291162ab384a70517874de6a4e24f2eafb550c7f4bdee477f0R56